stream-platform/src/main/java/io/camunda/zeebe/stream/impl/NoSuchProcessorException.java

```diff
@@ -16,7 +16,20 @@
  */
 final class NoSuchProcessorException extends UnrecoverableException {
 
-  NoSuchProcessorException(final TypedRecord<?> command) {
-    super("No processor registered for command type %s".formatted(command.getValueType()));
+  private NoSuchProcessorException(final String message) {
+    super(message);
+  }
+
+  public static NoSuchProcessorException forRecord(final TypedRecord<?> record) {
+    final var message =
+        switch (record.getRecordType()) {
+          case EVENT ->
+              "No processor registered for event type %s".formatted(record.getValueType());
```

Comment:
❌ 🤔 There are no processor for events (per definition) and also no for command rejections.

Reply:
This is the terminology we use within the stream-platform module. Everything is a `RecordProcessor`: https://github.com/camunda/zeebe/blob/4e17f4375c77b5087b7d1ae447e19f7fff2d18f3/stream-platform/src/main/java/io/camunda/zeebe/stream/api/RecordProcessor.java

Reply:
Ah so you referring to the record processor here? 

Still there will never be an exception for the command rejection, as processing will filter this already before.

Reply:
Yeah that's fair. I provided a special error message for `COMMAND_REJECTION`, `SBE_UNKNOWN`, and `NULL_VAL`. While it's currently not necessary because it should be filtered out, I think it's better to be safe here and also handle these cases so that we get nice error messages if we ever break the filtering.